File: Level-1/24.java
class Solution {
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        int largeVal;
        if(n < m){
            largeVal = m;
        }else{
            largeVal = n;
        }
        
        for(int i = largeVal; 1 < largeVal; i--){
            if(n % i == 0 && m % i == 0){
                answer[0] = i;
                answer[1] = n * m / i;
                break;
            }
        }
        
        return answer;
    }
}

File: Level-3/24.js
function solution(tickets) {
    const answer = [];
    const visited = [];
    
    tickets.sort();
    
    const len = tickets.length;
    const dfs = (str, count, path) => {
        if(count === len) {
            answer.push([...path, str]);
            return;
        }
        
        for(let i = 0; i < len; i++) {
            if(!visited[i] && tickets[i][0] === str) {
                visited[i] = true;
                dfs(tickets[i][1], count + 1, [...path, str]);
                visited[i] = false;
            }
        }
    };
    
    dfs('ICN', 0, []);
    
    return answer[0];
}
